src/ripper/ripcddialog.cpp

```diff
@@ -128,9 +131,15 @@ RipCDDialog::RipCDDialog(QWidget* parent)
       break;
     }
   }
+
+  // Start thread for polling
+  mediaChangedWatchingThread_.start();
 }
 
-RipCDDialog::~RipCDDialog() {}
+RipCDDialog::~RipCDDialog() {
+  QMutexLocker l(&mutex_);  // ensure now one is holding a lock on the mutex
+                            // when destroying..
```

Comment:
The thread locks the mutex multiple times per cycle, so it may call this class's methods after this. The Qt child objects aren't destroyed until the base QObject destructor is called. It would be safer to tell the thread to exit and wait for it here.